src/FileFormats/format_flatbuffer_opl3.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+ * OPL Bank Editor by Wohlstand, a free tool for music bank editing
+ * Copyright (c) 2016-2018 Vitaly Novichkov <admin@wohlnet.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "format_flatbuffer_opl3.h"
+#include "../common.h"
+#include "Opl3Bank_generated.h"
+
+bool FlatbufferOpl3::detect(const QString &, char *magic)
+{
+    return true;
+}
+
+FfmtErrCode FlatbufferOpl3::loadFile(QString filePath, FmBank &bank)
+{
+    uint16_t count_melodic_banks     = 0;
+    uint16_t count_percusive_banks   = 0;
+
+    QFile file(filePath);
+    if(!file.open(QIODevice::ReadOnly))
+        return FfmtErrCode::ERR_NOFILE;
+    
+    QByteArray blob = file.readAll();
+    file.close();
+
+    // Get a pointer to the root object inside the buffer.
+    auto opl3Bank = GetOpl3Bank(blob.constData());
+    auto banks = opl3Bank->banks();
+
+    // count banks types
+    for (uint8_t i = 0; i < banks->Length(); i++) {
+        auto bnk = banks->Get(i);
+        switch (bnk->type()) {
+            case BankType_Melodic:
+                count_melodic_banks++;
+                break;
+            case BankType_Percussion:
+                count_percusive_banks++;
+                break;
+            default:
+                return FfmtErrCode::ERR_BADFORMAT;
+        }
+    }
+
+    bank.reset(count_melodic_banks, count_percusive_banks);
+
+    bank.deep_vibrato = ((opl3Bank->oplTV() >> 0) & 0x01);
+    bank.deep_tremolo = ((opl3Bank->oplTV() >> 1) & 0x01);
+    switch (opl3Bank->volumeModel()) {
+        case VolumeModel_Auto:
+            bank.volume_model = 0;
+            break;
+        case VolumeModel_Generic:
+            bank.volume_model = 1;
+            break;
+        case VolumeModel_Native:
+            bank.volume_model = 2;
+            break;
+        case VolumeModel_DMX:
+            bank.volume_model = 3;
+            break;
+        case VolumeModel_Apogee:
+            bank.volume_model = 4;
+            break;
+        case VolumeModel_Win9x:
+            bank.volume_model = 5;
+            break;
+        default:
+            bank.volume_model = 0;
+            break;        
+    }
+    bank.volume_model = opl3Bank->volumeModel();
```

Comment:
Switch to check each value, and then... assign switch's input value as-is?
I bet, this totally kills a sense of entire `switch{}` block.